### PR TITLE
fix: always return count of workspaces

### PR DESCRIPTION
**Pull request by mtojek**
Related: https://github.com/coder/coder/issues/11795
Related: https://github.com/coder/coder/pull/12380

This PR ensures that `/workspaces` API endpoint always returns a count with the total number of workspaces. This is a simpler and alternative version to https://github.com/coder/coder/pull/12380.

Once it is merged, @BrunoQuaresma can work on the final adjustments on the FE side.

### CODE REVIEW

File: coderd/database/dbmem/dbmem.go [hunk 1]
```diff
@@ -389,6 +389,12 @@ func (q *FakeQuerier) convertToWorkspaceRowsNoLock(ctx context.Context, workspac
 
 		rows = append(rows, wr)
 	}
+	if withSummary {
+		rows = append(rows, database.GetWorkspacesRow{
+			Name:  "**TECHNICAL_ROW**",
+			Count: count,
+		})
+	}
```
Comment (johnstcn) [inline coderd/database/dbmem/dbmem.go:397]: I'm not 100% sold on the technical summary row. While it's not strictly compatibility-breaking, it's something that older clients won't know how to handle properly.

That having been said, given that it's an opt-in flag, it's not as big a deal.

**Comment by BrunoQuaresma** [comment]
Which adjustments do you think would be necessary?

**Review by johnstcn — COMMENTED** [review]
~I'm not 100% sold on the technical summary row. While it's not strictly compatibility-breaking, it's something that older clients won't know how to handle properly.~

~That having been said, if it can be an opt-in flag, then it's not a problem at all. Maybe we add a `&withSummary=true` query parameter to the `/api/v2/workspaces` endpoint as well and plumb that through? We can make it the default after a couple of releases.~

EDIT: I misinterpreted -- this is all happening on the backend, so clients don't come into it at all. I think this is fine then :+1:

**Review by johnstcn — APPROVED** [review]

**Comment by mtojek** [comment]
> That having been said, if it can be an opt-in flag, then it's not a problem at all. Maybe we add a &withSummary=true query parameter to the /api/v2/workspaces endpoint as well and plumb that through? We can make it the default after a couple of releases.

Oh, but API does not leak the technical row, right? From API perspective the only change is the "count" value. Is that what you meant?

**Comment by mtojek** [comment]
@BrunoQuaresma 

Basically this:

>  or even the right count for the number of workspaces so we can check if a pagination is wrong by doing some extra calculations.

from [your comment](https://github.com/coder/coder/issues/11795#issuecomment-1916674050)

**Comment by johnstcn** [comment]
> Oh, but API does not leak the technical row, right? From API perspective the only change is the "count" value. Is that what you meant?

Yeah, I was wondering if the technical row would show up as a "dummy" workspace in the API response. But that does not appear to be the case, correct?

**Comment by mtojek** [comment]
That would be a developer error, and there is no point in doing this.

**Pull request merged**